x-pack/plugins/cases/common/utils/user_actions.ts

```diff
@@ -5,14 +5,62 @@
  * 2.0.
  */
 
-export function isCreateConnector(action?: string, actionFields?: string[]): boolean {
-  return action === 'create' && actionFields != null && actionFields.includes('connector');
-}
+import {
+  ActionTypes,
+  CommentUserAction,
+  ConnectorUserAction,
+  CreateCaseUserAction,
+  DescriptionUserAction,
+  PushedUserAction,
+  StatusUserAction,
+  TagsUserAction,
+  TitleUserAction,
+  UserActionTypes,
+} from '../api';
+import { SnakeToCamelCase } from '../types';
 
-export function isUpdateConnector(action?: string, actionFields?: string[]): boolean {
-  return action === 'update' && actionFields != null && actionFields.includes('connector');
-}
+type SnakeCaseOrCamelCaseUserAction<
+  T extends 'snakeCase' | 'camelCase',
+  S,
+  C
+> = T extends 'snakeCase' ? S : C;
 
-export function isPush(action?: string, actionFields?: string[]): boolean {
-  return action === 'push-to-service' && actionFields != null && actionFields.includes('pushed');
-}
+export const isConnectorUserAction = (userAction: unknown): userAction is ConnectorUserAction =>
```

Comment:
Yes, this will be called with the full attributes (response type) but the problem is that on the front end the fields are in camel case and on the backend the fields are in snake case. The `is` will return true only for the snake case and it will fail on the front end.